WordPress/gutenberg
# feat: Separate pre-publish and post-publish panels visibility and settings

## What?
This PR addresses the[ issue](https://github.com/WordPress/gutenberg/issues/57268) of confusion in the publishing flow where the user interface displays a checkbox in the post-publish panel. The checkbox, when unchecked, disables both the pre and post-publish panels, affecting the user experience.

## Why?
Unchecking the only checkbox present disables both pre and post-publish panels, affecting the user's ability to receive suggestions and actions after publishing. The proposed changes aim to improve user understanding and provide a better user experience.

## How?
The PR proposes to introduce two different settings to toggle their visibility. This separation allows users to control the display of each panel independently, providing a clearer and more intuitive experience. The changes also ensure that unchecking the checkbox in the post-publish panel only affects the post-publish suggestions, not both panels.

## Testing Instructions
1. Open a post or page in the editor.
2. Navigate to the publishing flow.
3. Observe the presence and functionality of the "Always show pre-publish checks" checkbox in both pre and post-publish panels.
4. Check and uncheck the checkbox to verify its impact on the visibility of pre and post-publish panels.

### Testing Instructions for Keyboard
Navigate through the publishing flow using the keyboard and perform the above steps to ensure accessibility and keyboard-only testing.

## Screenshots

## Screenshots or screencast 
<img width="280" alt="Screenshot 2024-01-01 at 7 05 59 PM" src="https://github.com/WordPress/gutenberg/assets/90276347/bd46269e-c780-4f41-839b-dccda720c4a0">
